lib/rules/no-mutation-props.js

```diff
@@ -0,0 +1,222 @@
+/**
+ * @fileoverview Prevent mutation of this.props
+ * @author Ian Schmitz, Joey Baker
+ */
+'use strict';
+
+const Components = require('../util/Components');
+const ARRAY_MUTATIONS = ['push', 'pop', 'shift', 'unshift'];
```

Comment:
although i don't think we can safely check for any of these, you're also missing the other array mutator methods: `splice`, `reverse`, `sort`, `fill`, `copyWithin`.

Reply:
Added, but I'll hold of on adding tests until we resolve what to do here.